src/components/RouteSchedule/RouteSchedule.js

```diff
@@ -0,0 +1,246 @@
+/**
+ * Returns a color class to display the delay.
+ * @param {Number} time Delay time in milliseconds.
+ */
+const getDelayString = t => {
+  const h = Math.floor(t / 3600000);
+  const m = Math.floor((t % 3600000) / 60000);
+  const s = Math.floor(((t % 3600000) % 60000) / 1000);
+
+  if (s === 0 && h === 0 && m === 0) {
+    return '0';
+  }
+  if (s === 0 && h === 0) {
+    return `${pad(m)}m`;
+  }
+  if (s === 0) {
+    return `${pad(h)}h${pad(m)}m`;
+  }
+  if (m === 0 && h === 0) {
+    return `${pad(s)}s`;
+  }
+  if (h === 0) {
+    return `${pad(m)}m${pad(s)}s`;
+  }
+  return `${pad(h)}h${pad(m)}m${pad(s)}s`;
+};
+
+/**
+ * Returns a color class to display the delay.
+ * @param {Number} time Delay time in milliseconds.
+ */
+const getDelayColor = time => {
+  const secs = Math.round(((time / 1800 / 2) * 3600) / 1000);
+  if (secs >= 3600) {
+    return 'dark-red';
+  }
+  if (secs >= 500) {
+    return 'middle-red';
+  }
+  if (secs >= 300) {
+    return 'light-red';
+  }
+  if (secs >= 180) {
+    return 'orange';
+  }
+  return 'green';
+};
+
+/**
+ * Returns an image for first, middle or last stations.
+ * @param {Number} index index of the station in the list.
+ * @param {Number} length Length of the stations list.
+ */
+const renderStationImg = (index, length) => {
+  if (index === 0) {
+    return (
+      <img
+        src={firstStation}
+        alt="routeScheduleLine"
+        className="rt-route-icon"
+      />
+    );
+  }
+  if (index === length - 1) {
+    return (
+      <img
+        src={lastStation}
+        alt="routeScheduleLine"
+        className="rt-route-icon"
+      />
+    );
+  }
+  return (
+    <img src={station} alt="routeScheduleLine" className="rt-route-icon" />
+  );
+};
+
+const propTypes = {
+  /**
+   * CSS class of the route schedule wrapper.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Trajectory stations informations.
+   */
+  lineInfos: PropTypes.shape({
+    backgroundColor: PropTypes.string,
+    bicyclesAllowed: PropTypes.number,
+    color: PropTypes.string,
+    destination: PropTypes.string,
+    feedsId: PropTypes.number,
+    id: PropTypes.number,
+    longName: PropTypes.string,
+    operatingInformations: PropTypes.object,
+    operator: PropTypes.string,
+    operatorTimeZone: PropTypes.string,
+    operatorUrl: PropTypes.string,
+    realTime: PropTypes.number,
+    shortName: PropTypes.string,
+    stations: PropTypes.array,
+    vehiculeType: PropTypes.number,
+    wheelchairAccessible: PropTypes.number,
+  }),
+
+  /**
+   * Render Header of the route scheduler.
+   */
+  header: PropTypes.func,
+
+  /**
+   * Render Body of the route scheduler.
+   */
+  stations: PropTypes.func,
+
+  /**
+   * Function triggered on station's click event.
+   */
+  onStationClick: PropTypes.func,
+};
+
+const defaultProps = {
+  className: 'rt-route-wrapper',
+  lineInfos: null,
+  header: null,
+  stations: null,
+  onStationClick: () => {},
+};
+
+const renderHeader = lineInfos => {
```

Comment:
Please directly do `const renderHeader = lineInfos => (<div>...</div>)`
Btw.: My linter always complains about this. How do you trick your's? :stuck_out_tongue_winking_eye: